internal/utils/afterfunc.go

```diff
@@ -0,0 +1,36 @@
+//go:build !go1.21
+// +build !go1.21
+
+package utils
+
+import (
+	"context"
+	"sync/atomic"
+)
+
+func AfterFunc(ctx context.Context, f func()) (stop func() bool) {
```

Comment:
This implementation seems reasonable to me but can you elaborate on how you got to it? If I go to the source code for AfterFunc in the Go repo I see something that looks different and I want to make sure I understand why.

Reply:
Tests added, now we can verify that it conforms to the official requirements per the context.AfterFunc documentation.

Reply:
Thanks, the tests are great to have!